### ISSUE TITLE
Add new datetime column and make date just contain the date

### ISSUE THREAD

**Issue by Robinlovelace** [juanfonsecaLS1/telraamR#5]

**Comment by juanfonsecaLS1** [comment]
date and time should be corrected to reflect the time zone of the sensor

**Comment by juanfonsecaLS1** [comment]
solved by commit 4c0f6be

**Comment by temospena** [comment]
Hi,
Thanks for this pkg :)

I was trying to do the README examples with my data, and I think that after this commit, now the dir_data (as in the example README) brings `datetime` and not `date`. This is fine, but in the plot example, it needs one more line to work:

``` r
dir_data |>
  mutate(date = lubridate::as_date(datetime)) |> # create date, otherwise factor(date) won't work
  filter(type=="car") |> 
  ggplot(aes(x=hr,y=flow,col=road_dir))+
  geom_line(aes(group=factor(date)),alpha = 0.2)+ # date does not exist in dir_data (only datetime)
  stat_summary(fun = median,geom = "line",linewidth=1)+
  facet_grid(.~road_dir)+
  labs(x = "Hour", y = "Cars per hour")+
  theme(legend.position = "none")
```

**Comment by juanfonsecaLS1** [comment]
Hi @temospena, you are welcome, I hope it is useful. I just tested the latest version of the package and it does have both `date` and `datetime`. Is it the case that  `tidy_data` is not returning the `date` column or the `data` object did not have it at all?

**Comment by temospena** [comment]
`tidy_directional` is not returning the `date` column (only `datetime`)

Here goes a reprex:

``` r
library(telraamR)
library(tidyverse)

data = read_telraam_traffic(9000005488, #mine
                            time_start = "2023-08-01 07:00:00",
                            time_end = "2023-09-25 07:00:00",
                            report = "per-hour",
                            include_speed = TRUE)
#> Warning in read_telraam_traffic(9000005488, time_start = "2023-08-01 07:00:00",
#> : End time is in the future. End time constained to current system time

dir_data = data |>
  tidy_directional()
#> Linking to GEOS 3.11.2, GDAL 3.6.2, PROJ 9.2.0; sf_use_s2() is TRUE

names(dir_data)
#> [1] "segment_id" "uptime"     "timezone"   "datetime"   "day"       
#> [6] "hr"         "road_dir"   "type"       "flow"

dir_data |>
  filter(type=="car") |> 
  ggplot(aes(x=hr,y=flow,col=road_dir))+
  geom_line(aes(group=factor(date)),alpha = 0.2)+
  stat_summary(fun = median,geom = "line",linewidth=1)+
  facet_grid(.~road_dir)+
  labs(x = "Hour", y = "Cars per hour")+
  theme(legend.position = "none")
#> Error in `geom_line()`:
#> ! Problem while computing aesthetics.
#> ℹ Error occurred in the 1st layer.
#> Caused by error in `unique.default()`:
#> ! unique() applies only to vectors
#> Backtrace:
#>      ▆
#>   1. ├─base::tryCatch(...)
#>   2. │ └─base (local) tryCatchList(expr, classes, parentenv, handlers)
#>   3. │   ├─base (local) tryCatchOne(...)
#>   4. │   │ └─base (local) doTryCatch(return(expr), name, parentenv, handler)
#>   5. │   └─base (local) tryCatchList(expr, names[-nh], parentenv, handlers[-nh])
#>   6. │     └─base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
#>   7. │       └─base (local) doTryCatch(return(expr), name, parentenv, handler)
#>   8. ├─base::withCallingHandlers(...)
#>   9. ├─base::saveRDS(...)
#>  10. ├─base::do.call(...)
#>  11. ├─base (local) `<fn>`(...)
#>  12. ├─global `<fn>`(input = base::quote("dopey-nyala_reprex.R"))
#>  13. │ └─rmarkdown::render(input, quiet = TRUE, envir = globalenv(), encoding = "UTF-8")
#>  14. │   └─knitr::knit(knit_input, knit_output, envir = envir, quiet = quiet)
#>  15. │     └─knitr:::process_file(text, output)
#>  16. │       ├─base::withCallingHandlers(...)
#>  17. │       ├─base::withCallingHandlers(...)
#>  18. │       ├─knitr:::process_group(group)
#>  19. │       └─knitr:::process_group.block(group)
#>  20. │         └─knitr:::call_block(x)
#>  21. │           └─knitr:::block_exec(params)
#>  22. │             └─knitr:::eng_r(options)
#>  23. │               ├─knitr:::in_input_dir(...)
#>  24. │               │ └─knitr:::in_dir(input_dir(), expr)
#>  25. │               └─knitr (local) evaluate(...)
#>  26. │                 └─evaluate::evaluate(...)
#>  27. │                   └─evaluate:::evaluate_call(...)
#>  28. │                     ├─evaluate (local) handle(...)
#>  29. │                     │ └─base::try(f, silent = TRUE)
#>  30. │                     │   └─base::tryCatch(...)
#>  31. │                     │     └─base (local) tryCatchList(expr, classes, parentenv, handlers)
#>  32. │                     │       └─base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
#>  33. │                     │         └─base (local) doTryCatch(return(expr), name, parentenv, handler)
#>  34. │                     ├─base::withCallingHandlers(...)
#>  35. │                     ├─base::withVisible(value_fun(ev$value, ev$visible))
#>  36. │                     └─knitr (local) value_fun(ev$value, ev$visible)
#>  37. │                       └─knitr (local) fun(x, options = options)
#>  38. │                         ├─base::withVisible(knit_print(x, ...))
#>  39. │                         ├─knitr::knit_print(x, ...)
#>  40. │                         └─knitr:::knit_print.default(x, ...)
#>  41. │                           └─evaluate (local) normal_print(x)
#>  42. │                             ├─base::print(x)
#>  43. │                             └─ggplot2:::print.ggplot(x)
#>  44. │                               ├─ggplot2::ggplot_build(x)
#>  45. │                               └─ggplot2:::ggplot_build.ggplot(x)
#>  46. │                                 └─ggplot2:::by_layer(...)
#>  47. │                                   ├─rlang::try_fetch(...)
#>  48. │                                   │ ├─base::tryCatch(...)
#>  49. │                                   │ │ └─base (local) tryCatchList(expr, classes, parentenv, handlers)
#>  50. │                                   │ │   └─base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
#>  51. │                                   │ │     └─base (local) doTryCatch(return(expr), name, parentenv, handler)
#>  52. │                                   │ └─base::withCallingHandlers(...)
#>  53. │                                   └─ggplot2 (local) f(l = layers[[i]], d = data[[i]])
#>  54. │                                     └─l$compute_aesthetics(d, plot)
#>  55. │                                       └─ggplot2 (local) compute_aesthetics(..., self = self)
#>  56. │                                         └─ggplot2:::scales_add_defaults(...)
#>  57. │                                           └─base::lapply(aesthetics[new_aesthetics], eval_tidy, data = data)
#>  58. │                                             └─rlang (local) FUN(X[[i]], ...)
#>  59. ├─base::factor(date)
#>  60. │ ├─base::unique(x, nmax = nmax)
#>  61. │ └─base::unique.default(x, nmax = nmax)
#>  62. └─base::.handleSimpleError(...)
#>  63.   └─rlang (local) h(simpleError(msg, call))
#>  64.     └─handlers[[1L]](cnd)
#>  65.       └─cli::cli_abort(...)
#>  66.         └─rlang::abort(...)

dir_data |>
  mutate(date = lubridate::as_date(datetime)) |> # add date
  filter(type=="car") |> 
  ggplot(aes(x=hr,y=flow,col=road_dir))+
  geom_line(aes(group=factor(date)),alpha = 0.2)+
  stat_summary(fun = median,geom = "line",linewidth=1)+
  facet_grid(.~road_dir)+
  labs(x = "Hour", y = "Cars per hour")+
  theme(legend.position = "none")
```

![](https://i.imgur.com/kbTazw0.png)

<sup>Created on 2023-09-24 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by juanfonsecaLS1** [comment]
Reopening this issue. `date` is missing when applying the `tidy_directional` function

**Comment by juanfonsecaLS1** [comment]
Fixed in commit 76c4d14. Thank you @temospena for spotting this.
